WireAPI/Sources/WireAPI/APIs/UpdateEventsAPI/UpdateEvent+Federation.swift

```diff
@@ -24,13 +24,57 @@ extension UpdateEvent {
         eventType: FederationEventType,
         from decoder: any Decoder
     ) throws {
+        let container = try decoder.container(keyedBy: FederationEventCodingKeys.self)
+
         switch eventType {
         case .connectionRemoved:
-            self = .federation(.connectionRemoved)
+            let event = try container.decodeConnectionRemovedEvent()
+            self = .federation(.connectionRemoved(event))
 
         case .delete:
-            self = .federation(.delete)
+            let event = try container.decodeDeleteEvent()
+            self = .federation(.delete(event))
         }
     }
 
 }
+
+private enum FederationEventCodingKeys: String, CodingKey {
+
+    case payload = "data"
+
+}
+
+// MARK: - Federation connection removed
+
+private extension KeyedDecodingContainer<FederationEventCodingKeys> {
+
+    func decodeConnectionRemovedEvent() throws -> FederationConnectionRemovedEvent {
+        let payload = try decode(FederationConnectionRemovedEventPayload.self, forKey: .payload)
+        return FederationConnectionRemovedEvent(domains: payload.domains)
+    }
+
+    private struct FederationConnectionRemovedEventPayload: Decodable {
+
+        let domains: Set<String>
+
+    }
+
+}
```

Comment:
Hm should this be an extension with a single func used in a single place? We can do it also without extension quite simple:

```swift
    // switch 
    case .connectionRemoved:
            let payload = try container.decode(FederationConnectionRemovedEventPayload.self, forKey: .payload)
            let event = FederationConnectionRemovedEvent(domains: payload.domains)

// ...

private struct FederationConnectionRemovedEventPayload: Decodable {

    let domains: Set<String>

}
```

Reply:
When I started with the conversation events, there are a lot of events to decode, and often the event payloads contain more than a simple payload. To avoid having a super long switch and to try improve readability, I created these private extensions, one for each event type.

There are only two federation events, so it may seem unnecessary here, but I kept the same approach to be consistent. I'm happy to also do it inline here.

Reply:
I see, consistency is also important so either way is fine for me.